app/timeline-menu/timeline-menu.ts

```diff
@@ -0,0 +1,63 @@
+let _ = require('lodash');
+
+export enum TimelineMenuRange { LastWeek, LastMonth, LastYear, LastDecade, PreDecade, CustomRange }
+
+export class TimelineMenu {
+  LIMIT_NUM = 8640000000000000;
+  MIN_DATE = new Date(-this.LIMIT_NUM);
+  MAX_DATE = new Date(this.LIMIT_NUM);
+  periods = [
+    { 'title': '7 ימים אחרונים', 'value': 'LastWeek', 'start': '', 'end': '' },
+    { 'title': '30 ימים אחרונים', 'value': 'LastMonth', 'start': '', 'end': '' },
+    { 'title': 'השנה', 'value': 'LastYear', 'start': '', 'end': '' },
+    { 'title': 'בעשור האחרון', 'value': 'LastDecade', 'start': '', 'end': '' },
+    { 'title': 'מלפני עשור ויותר', 'value': 'PreDecade', 'start': '', 'end': '' },
+    { 'title': 'טווח תאריכים לבחירתך', 'value': 'CustomRange', 'start': '', 'end': '' }
+  ];
+
+  constructor() {
+    this.initPeriods();
+  }
+
+  getPeriod(range: TimelineMenuRange) {
+    return _.find(this.periods, ['value', TimelineMenuRange[range]]);
+  }
+
+  private initPeriods(): any {
+    let that = this;
+    _.forEach(this.periods, function(value: any, key: any) {
```

Comment:
Very cumbersome initialization method.

Just create the period objects with the values inside...

Reply:
the period object is object which contains dates and dates are different each time when we check it/request budget. Last week for example is different for yesterday and for tomorrow...